extension/entitystore/retryer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"go.uber.org/zap"
+)
+
+const (
+	RequestLimitExceeded = "RequestLimitExceeded"
+	infRetry             = -1
+)
+
+var (
+	retryableErrorMap = map[string]bool{
+		"RequestLimitExceeded": true,
+	}
+)
+
+type Retryer struct {
+	oneTime         bool
+	retryAnyError   bool
+	successRetryMin int
+	successRetryMax int
+	backoffArray    []time.Duration
+	maxRetry        int
+	done            chan struct{}
+	logger          *zap.Logger
+}
+
+func NewRetryer(onetime bool, retryAnyError bool, successRetryMin int, successRetryMax int, backoffArray []time.Duration, maxRetry int, done chan struct{}, logger *zap.Logger) *Retryer {
+	return &Retryer{
+		oneTime:         onetime,
+		retryAnyError:   retryAnyError,
+		successRetryMin: successRetryMin,
+		successRetryMax: successRetryMax,
+		backoffArray:    backoffArray,
+		maxRetry:        maxRetry,
+		done:            done,
+		logger:          logger,
+	}
+}
+
+func (r *Retryer) refreshLoop(updateFunc func() error) int {
```

Comment:
nit: Why does this return an int if it's never used?

Reply:
Its for unit testing purpose